### PR TITLE
[WIP] Add a very simple overview of secrets

### PR DESCRIPTION
**Pull request by LucasRoesler**
**What**
- Add new Reference section
- Add a document on using generic secrets to the Reference section

**Notes**
- This current version is to get initial feedback on the style and organization

### CODE REVIEW

File: docs/reference/secrets.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+From the command line use
+
+```sh
+docker secret create secret_api_key ~/secrets/secret_api_key.txt
+```
+
+## Use the secret in your function
+
+This is the simplest part, update your stack file to include the secret:
```
Comment (alexellis) [inline docs/reference/secrets.md]: I'd drop "This is the simplest part"

File: docs/reference/secrets.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+         image: functions/api-key-protected:latest
+         secrets:
+          - secret_api_key
+    ```
+
+and then deploy `faas-cli deploy -f ./stack.yaml`
+
+Once the deploy is done you can test the function using
+
+```sh
```
Comment (alexellis) [inline docs/reference/secrets.md]: I like the brevity here 💯 - can you just explain what the function is doing and show expected output with a correct header and an incorrect header? (Do this by testing it out on either Swarm or K8s)

File: docs/reference/secrets.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Using secrets
+
+This page shows how to use secrets within your functions for API tokens, passwords and similar.
+
+Using secrets is a two step process. First we need to define the secret in your cluster and then you need to 'use' the secret to your function. You can find a simple example function [ApiKeyProtected in the OpenFaaS repo](https://github.com/openfaas/faas/tree/master/sample-functions/ApiKeyProtected-Secrets). When we deploy this function we provide a secret key that it uses to authenticate requests.
+
+_Note_: The examples in the following section require `faas-cli` version `>=0.5.1`
```
Comment (alexellis) [inline docs/reference/secrets.md]: Good note. Was this added historically? Is it now safe to assume people are already on this version?

Comment (LucasRoesler) [reply docs/reference/secrets.md]: I pulled a lot of this from the secrets walkthrough in the faas repo, we added it awhile back.  I honestly don't know if it is needed, I would be surprised if people were using these docs and not already on a compatible version.  I figured it was safest to start including it and then decide if we should remove it

**Comment by LucasRoesler** [comment]
@alexellis I made adjustments for each of your comments

**Comment by alexellis** [comment]
Thanks for a quick turn-around!

**Comment by alexellis** [comment]
Merged

**Pull request merged**